lib/selenium.pm

```diff
@@ -0,0 +1,198 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017-2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Support for Selenium
+# Maintainer: Ondrej Holecek oholecek@suse.com>
+
+package selenium;
+use 5.018;
+
+use base 'Exporter';
+use Exporter;
+
+our @EXPORT = qw(
+  add_chromium_repos
+  install_chromium
+  enable_selenium_port
+  selenium_driver
+  wait_for_page_to_load
+  wait_for_link
+  wait_for_text
+  wait_for_xpath
+  select_input
+);
+
+use testapi;
+use utils 'zypper_call';
+use opensusebasetest;
+use version_utils 'is_sle';
+
+use Selenium::Remote::Driver;
+use Selenium::Chrome;
+use Selenium::Waiter qw/wait_until/;
+use Selenium::Remote::WDKeys;
+
+my $port = 4444;
+
+=head1 openQA - selenium webdriver support
+
+This Selenium module exports subroutines helpers to integrate C<Selenium::Remote::Driver>
+into openQA tests
+
+Note: Selenium module works only when openvswitch networking is used!
+
+Chromedriver and chromium are installed on SUT, firewall is enabled. OpenQA worker then initiate
+connection to SUT and returns C<Selenium::Remote::Driver> to be used in tests.
+
+Usage:
+
+  use selenium;
+  add_chromium_repos;   # for SLES12
+
+  install_chromium;
+  enable_selenium_port;
+
+  my $driver = selenium_driver;
+  ...
+=cut
+sub add_chromium_repos {
+    my $ret = zypper_call("se chromedriver", exitcode => [0, 104]);
+    if ($ret == 104) {
+        if (is_sle('<15')) {
+            zypper_call(
+                '--gpg-auto-import-keys ar -fc http://download.opensuse.org/repositories/openSUSE:/Backports:/SLE-12/standard/openSUSE:Backports:SLE-12.repo');
+            zypper_call(
+'--gpg-auto-import-keys ar -fc http://download.opensuse.org/repositories/openSUSE:/Backports:/SLE-12-SP1/standard/openSUSE:Backports:SLE-12-SP1.repo'
+            );
+            zypper_call(
+'--gpg-auto-import-keys ar -fc http://download.opensuse.org/repositories/openSUSE:/Backports:/SLE-12-SP2/standard/openSUSE:Backports:SLE-12-SP2.repo'
+            );
+            zypper_call(
+'--gpg-auto-import-keys ar -fc http://download.opensuse.org/repositories/openSUSE:/Backports:/SLE-12-SP3/standard/openSUSE:Backports:SLE-12-SP3.repo'
+            );
+        }
+        elsif (is_sle('>=15')) {
+            zypper_call(
+                '--gpg-auto-import-keys ar -fc http://download.opensuse.org/repositories/openSUSE:/Backports:/SLE-15/standard/openSUSE:Backports:SLE-15.repo');
+        }
+        zypper_call('--gpg-auto-import-keys ref');
+    }
+}
+
+sub install_chromium {
+    zypper_call('in chromium chromedriver');
+}
+
+sub enable_selenium_port {
+    assert_script_run('systemctl stop ' . opensusebasetest::firewall());
```

Comment:
Could be replace by `systemctl 'stop ' . firewall;.` (and `opensusebasetest::` is not needed because you already include that librairie).

Reply:
I'm not sure about this one. opensusebasetest does not inherit import from exporter so it does not provide firewall as is. Or am I mistaken? I did not try it without prefix.

Reply:
For me it should work, but you can have a try later and correct this after, yes.